bgpd/bgp_routemap.c

```diff
@@ -2348,29 +2348,52 @@ route_set_aspath_replace(void *rule, const struct prefix *dummy, void *object)
 	struct aspath *aspath_new;
 	const char *replace = rule;
 	struct bgp_path_info *path = object;
-	as_t own_asn = path->peer->change_local_as ? path->peer->change_local_as
-						   : path->peer->local_as;
+	as_t replace_asn;
+	as_t configured_asn;
+	char *buf;
+	char src_asn[ASN_STRING_MAX_SIZE];
 
 	if (path->peer->sort != BGP_PEER_EBGP) {
 		zlog_warn(
 			"`set as-path replace` is supported only for EBGP peers");
 		return RMAP_NOOP;
 	}
 
+	buf = strchr(replace, ' ');
+	if (!buf) {
+		configured_asn = path->peer->change_local_as
+					 ? path->peer->change_local_as
+					 : path->peer->local_as;
+	} else {
+		memcpy(src_asn, replace, (size_t)(buf - replace));
+		src_asn[(size_t)(buf - replace)] = '\0';
+		replace = src_asn;
+		buf++;
+		if (!asn_str2asn(buf, &configured_asn)) {
+			zlog_warn(
+				"`set as-path replace`, invalid configured as number %s",
```

Comment:
nit: as -> AS to avoid misreading `as number`

Reply:
done